src/compiler/collections.ts

```diff
@@ -222,8 +229,12 @@ namespace ts {
             this.data = createDictionaryModeObject<true>();
         }
 
-        delete(value: string) {
-            delete this.data[value];
+        delete(value: string): boolean {
+            const had = this.has(value);

```

Comment:
As above, you can just do `return delete this.data[value]` since a `delete` expression returns `boolean`